moshi/src/main/java/com/squareup/moshi/JsonAdapter.java

```diff
@@ -37,7 +37,9 @@
   }
 
   public final @Nullable T fromJson(String string) throws IOException {
-    return fromJson(new Buffer().writeUtf8(string));
+    try (Buffer buffer = new Buffer()) {
+    	return fromJson(buffer.writeUtf8(string));
+    }
```

Comment:
writing to a Buffer does not IO, and the changes outside of ClassJsonAdapter are unnecessary and unrelated to the change you are trying to make. will you revert these changes please?

Reply:
this way an accident with other code cleanup i like to submit later..

Reply:
anyway Buffer is anAutoClosabe and these should have to be close otherwise code checkers are argue about it